ts/features/walletV3/payment/screens/WalletPaymentPickMethodScreen.tsx

```diff
@@ -44,51 +161,171 @@ const WalletPaymentPickMethodScreen = () => {
-          disabled: isLoading || !canContinue,
-          loading: isLoading
-        }}
-      >
-        <DebugPrettyPrint title="paymentMethodsPot" data={paymentMethodsPot} />
-        <VSpacer size={16} />
-        <DebugPrettyPrint title="userWalletsPots" data={userWalletsPots} />
-      </GradientScrollView>
-    </BaseScreenComponent>
+// ----------------------- UTILS -----------------------
+
+const getIconWithFallback = (
+  brand?: string
+): ComponentProps<typeof ListItemRadio>["startImage"] => {
+  const logos = IOPaymentLogos;
+  return pipe(
+    brand,
+    O.fromNullable,
+    O.chain(findFirstCaseInsensitive(logos)),
+    O.map(([brand]) => brand),
+    O.fold(
+      () => ({ icon: "creditCard" }),
+      // @ts-expect-error ts whines because this function can return
+      // two different -- both correct -- types (see return type)
+      brand => ({ paymentLogo: brand as IOPaymentLogos })
+    )
   );
 };
+const mapGenericToRadioItem = (
+  method: PaymentMethodResponse,
+  transactionAmount?: number
+): RadioItem<string> => ({
+  id: method.id,
+  value: method.description,
+  disabled: isDisabled(method, transactionAmount),
+  startImage: getIconWithFallback(method.asset)
+});
+
+// should never return void, but since this is a map function it's expectable
+const mapSavedToRadioItem = (
+  method: WalletInfo
+): RadioItem<string> | undefined => {
+  switch (method.details?.type) {
+    case "CARDS":
+      const cardDetails = method.details as WalletInfoDetails1;
+      return {
+        id: method.walletId,
+        value: `${capitalize(cardDetails.brand)} ••${cardDetails.maskedPan}`,
+        startImage: getIconWithFallback(cardDetails.brand)
+      };
+    case "PAYPAL":
+      return {
+        id: method.walletId,
+        value: "PayPal",
+        startImage: getIconWithFallback("paypal")
+      };
+    case "BANCOMATPAY":
+      return {
+        id: method.walletId,
+        value: "BANCOMAT Pay",
+        startImage: getIconWithFallback("bancomatpay")
+      };
+    default:
+      return undefined;
+  }
+};
+
+// not sure if this ranges[0] thing is the right way, but
+// it's pretty easy to add full traversal, even though it
+// makes the code more complex
+const isDisabled = (
+  method: PaymentMethodResponse,
+  transactionAmount?: number
+): boolean =>
+  pipe(
+    sequenceS(O.Monad)({
+      min: O.fromNullable(method.ranges[0].min),
+      max: O.fromNullable(method.ranges[0].max),
+      amount: O.fromNullable(transactionAmount)
+    }),
+    O.fold(
+      () => false,
+      ({ min, max, amount }) => min > amount || max < amount
+    )
+  );
+
+const loadingRadios: Array<RadioItem<string>> = [
+  {
+    id: "1",
+    disabled: true,
+    loadingProps: { state: true },
+    value: "123456"
+  },
+
+  {
+    id: "2",
+    disabled: true,
+    loadingProps: { state: true },
+    value: "123456"
+  },
+  {
+    id: "3",
+    disabled: true,
+    loadingProps: { state: true },
+    value: "123456"
+  }
+];
```

Comment:
Following UI updates, this list now should be longer. I'd suggest to generate the list instead of write it manually, something like:

```suggestion
const loadingRadios: Array<RadioItem<string>> = Array.from(
  { length: 10 },
  (_, id) => ({
    id: id.toString(),
    disabled: true,
    loadingProps: { state: true },
    value: ""
  })
);
```

What do you think?

Reply:
way better, addressed 👍